main.rb

```diff
@@ -0,0 +1,229 @@
+# rubocop:disable Style/For
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    if instance_of?(Hash)
+      for i in 0..length - 1 do
+        yield keys[i], values[i]
+      end
+    else
+      for i in 0..to_a.length - 1 do
+        yield to_a[i]
+      end
+    end
+    self
+  end
```

Comment:
# my_each
- [ ] `my_each` when a block is given when self is a hash behaves like #each calls the given block once for each element in self

     Failure/Error: expect(my_each_output).to eq(each_output)

       expected: "[:a, 1][:b, 2][:c, 3][:d, 4][:e, 5]"
            got: "abcde"